### PR TITLE
plot data

### LINKED ISSUE #132: Add plots
[pyQtGraph](https://www.pyqtgraph.org/) or [Matplotlib](https://matplotlib.org/)?

### PR DESCRIPTION
**Pull request by prjemian**
- close #132

### CODE REVIEW

**Comment by prjemian** [comment]
Inspired by progress in [mdaviz](https://github.com/BCDA-APS/mdaviz/pull/29#issuecomment-1709253994).

This screen would be used to select data fields for 1-D line plotting from a single stream of a BlueskyRun as received from tiled.

image

The screen could be wrapped in a QTabWIdget, with one tab for each stream with plottable (numeric) data in the selected run.  Yet to be worked out...  Or a QComboBox could be used to select the stream with plottable (numeric) data.  I like the first because the internal data structures of this model would not need to be reconstructed if the stream is changed from a pop-up menu.

**Comment by prjemian** [comment]
Thinking to rename this model/view to something more generic, such as `XYMonTable` and `XYMonView`.  With a more general handling of column names, it could be used both here and in [mdaviz](https://github.com/BCDA-APS/mdaviz/pull/29#pullrequestreview-1618897367).

**Comment by prjemian** [comment]
Refer to each data to be plotted as a _field_, which avoids other names in use (such as _signal_).

image

**Comment by prjemian** [comment]
image

**Comment by prjemian** [comment]
Pre-selected columns, additional columns (of text):
image

**Comment by prjemian** [comment]
Plot selection (a transient activity) is better done in a modal dialog than a panel on the main window.

**Comment by prjemian** [comment]
Plot field selection dialog now:
image

**Comment by prjemian** [comment]
The first plot!
image

**Comment by prjemian** [comment]
@rodolakis : As we discussed, this PR is huge.  Since you have used the ChartView widget in mdaviz, review of that code is acceptable to some extent.  We'll add issues to modify from this point.

Please make a review so that this PR can be merged.  We'll come back to other issues later this week.

File: gemviz/select_fields_tablemodel.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+    def updateCheckboxes(self):
+        """Update checkboxes to agree with self.selections."""
+        if len(self.selections) > 0:
+            top, bottom = min(self.selections), max(self.selections)
+        else:
+            top, bottom = 0, self.rowCount() - 1
+        left, right = min(self.checkboxColumns), max(self.checkboxColumns)
+        logger.debug("corners: (%d,%d)  (%d,%d)", top, left, bottom, right)
+
+        # Re-evaluate the checkboxes bounded by the two corners (inclusive).
```
Comment (rodolakis) [inline gemviz/select_fields_tablemodel.py:251]: That is the only part of the code that I did not understand, what are you doing with the corners?

Comment (rodolakis) [reply gemviz/select_fields_tablemodel.py:251]: (I skipped it in mdaviz)

**Comment by prjemian** [comment]
The corners advise the QTableView widget what part of the table needs to be
redrawn. That way, out does not have to rebuild the entire table, a task
which might be expensive.

**Review by rodolakis — APPROVED** [review]
LGTM

**Pull request merged**